WcaOnRails/app/models/competition.rb

```diff
@@ -261,4 +285,25 @@ def end_date
       errors.add(:eventSpecs, "invalid event ids: #{invalid_events.map(&:id).join(',')}")
     end
   end
+
+  def competitions_nearby
+    Competition.where(
+      "ABS(DATEDIFF(?, CONCAT(year, '-', month, '-', day))) <= ? AND id <> ?", start_date, Competition.NEARBY_DAYS_WARNING, id)
+      .select { |c| kilometers_to(c) <= Competition.NEARBY_DISTANCE_KM_WARNING }
+      .sort_by { |c| kilometers_to(c) }
+  end
+
+  # Source http://www.movable-type.co.uk/scripts/latlong.html
+  def kilometers_to(c)
+    def to_radians(degrees)
```

Comment:
`to_radians` could be useful for `latitude_radians` and `longitude_radians` as well?